### PR TITLE
refactoring utility function in SelectReadOnlyText to create JeditoPane whith HyperLinkListenerAndToolTips

### PR DESCRIPTION
**Pull request by PPAC37**
Surely a little more greedy but seems much more pleasant / generic and user friendly. 
image
image

what do you think?

### CODE REVIEW

File: src/main/java/com/marginallyclever/makelangelo/select/SelectReadOnlyText.java [hunk 1]
```diff
@@ -2,56 +2,83 @@
...
+ * 
+ * @author Dan Royer (and modified by PPAC37)
  * @since 7.24.0
  */
 public class SelectReadOnlyText extends Select {
 	/**
 	 * 
 	 */
-	private static final long serialVersionUID = -8918068053490064344L;
-	private static final String A_HREF = "<a href=";
```
Comment (i-make-robots) [inline src/main/java/com/marginallyclever/makelangelo/select/SelectReadOnlyText.java:23]: will break if someone does something dumb like "<a     href    ="

File: src/main/java/com/marginallyclever/makelangelo/select/SelectReadOnlyText.java [hunk 2]
```diff
@@ -2,56 +2,83 @@
...
-	private static final String HREF_CLOSED = ">";
+	private static final long serialVersionUID = -8918068053490064344L;// TODO to regenerat ?
```
Comment (i-make-robots) [inline src/main/java/com/marginallyclever/makelangelo/select/SelectReadOnlyText.java:27]: this is never actually serialized, so I don't believe it needs to be regenerated.

File: src/main/java/com/marginallyclever/makelangelo/select/SelectReadOnlyText.java [hunk 3]
```diff
@@ -2,56 +2,83 @@
...
+	 * @param sToSetAsTextToTheHtmlEditorPane
+	 * @return 
+	 */
+	public static JEditorPane createJEditorPaneWithHyperlinkListenerAndToolTipsForDesktopBrowse(String sToSetAsTextToTheHtmlEditorPane) {
+		JEditorPane createdJEditorPane =  new JEditorPane();
+		createdJEditorPane.setEditable(false);
+		createdJEditorPane.setOpaque(false);
+		createdJEditorPane.setContentType("text/html");
+		createdJEditorPane.setText(sToSetAsTextToTheHtmlEditorPane);
+		createdJEditorPane.addHyperlinkListener(new HyperlinkListener() {
```
Comment (i-make-robots) [inline src/main/java/com/marginallyclever/makelangelo/select/SelectReadOnlyText.java:50]:
```
createdJEditorPane.addHyperlinkListener( (hyperlinkEvent)->{
```
maybe?

File: src/main/java/com/marginallyclever/makelangelo/select/SelectReadOnlyText.java [hunk 4]
```diff
@@ -2,56 +2,83 @@
...
+		jEdPane = createJEditorPaneWithHyperlinkListenerAndToolTipsForDesktopBrowse("<html>"+ labelKey+"</html>");
+		this.add(jEdPane,BorderLayout.CENTER);
+	}
+	
+	/**
+	 * Create a JEditorPane not editable for text/html contente, with an HyperLinkListener to Desktop Browse (when clicked) and show a ToolTips with the URL hovered.
+	 * @param sToSetAsTextToTheHtmlEditorPane
+	 * @return 
+	 */
+	public static JEditorPane createJEditorPaneWithHyperlinkListenerAndToolTipsForDesktopBrowse(String sToSetAsTextToTheHtmlEditorPane) {
```
Comment (i-make-robots) [inline src/main/java/com/marginallyclever/makelangelo/select/SelectReadOnlyText.java:44]: I love this. never be afraid to use long and very descriptive names.

File: src/main/java/com/marginallyclever/makelangelo/select/SelectReadOnlyText.java (hunk 1 above)
Comment (PPAC37) [reply src/main/java/com/marginallyclever/makelangelo/select/SelectReadOnlyText.java:23]: I hadn't paid attention, I guess with the JEditorPane and HyperlinkListener this should no longer be a problem.

File: src/main/java/com/marginallyclever/makelangelo/select/SelectReadOnlyText.java (hunk 4 above)
Comment (PPAC37) [reply src/main/java/com/marginallyclever/makelangelo/select/SelectReadOnlyText.java:43]: :) At the time I said to myself that I may be abusing ... but after the video on how to code well, I adhere to "the code must explain itself"

File: src/main/java/com/marginallyclever/makelangelo/select/SelectReadOnlyText.java (hunk 2 above)
Comment (PPAC37) [reply src/main/java/com/marginallyclever/makelangelo/select/SelectReadOnlyText.java]: Thanks, comment removed.

File: src/main/java/com/marginallyclever/makelangelo/select/SelectReadOnlyText.java (hunk 3 above)
Comment (PPAC37) [reply src/main/java/com/marginallyclever/makelangelo/select/SelectReadOnlyText.java]: Thank you I too often forget the lambda expressions. I have to revisit my habits.

**Pull request merged**